src/app/model.ts

```diff
@@ -1,36 +1,127 @@
-}
+import {AppFieldNames, ZDFieldValidation, getMultiselectValues} from '../utils';
+
+import {configStore} from '../store';
 
-export const getTicketFromForm = (values: CreateTicketFormValues): Tickets.CreatePayload => {
-    const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+export type FieldValidationErrors = {[name: string]: string};
 
-    const additionalMessage = values.additional_message || '';
-    const postMessage = values.post_message || '';
+interface ITicketFromFrom {
+    getTicket(): Tickets.CreatePayload;
+    fieldValidationErrors: FieldValidationErrors;
+}
 
-    const zdMessage = additionalMessage + '\n' +
-            postMessage + '\n' +
-            mmSignature;
+export class TicketFromForm implements ITicketFromFrom {
+    formValues: AppFormValues;
+    ticket: Tickets.CreateModel;
+    fieldValidationErrors: FieldValidationErrors
 
-    const ticket: Tickets.CreatePayload = {
-        ticket: {
-            subject: values.subject,
-            type: values.type,
-            priority: values.priority,
+    constructor(values: AppFormValues) {
+        this.formValues = values;
+        this.fieldValidationErrors = {};
+        this.ticket = {
             comment: {
-                body: zdMessage,
+                body: this.getPostMessage(),
             },
-        },
-    };
-    return ticket;
-};
+        };
+    }
+
+    getPostMessage(): string {
+        const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+
+        const additionalMessage = this.formValues[AppFieldNames.AdditionalMessage] || '';
+        const postMessage = this.formValues[AppFieldNames.PostMessage] || '';
+
+        const zdMessage = additionalMessage + '\n' +
+                postMessage + '\n' +
+                mmSignature;
+
+        return zdMessage;
+    }
+
+    getTicket(): Tickets.CreatePayload {
+        this.mapFormValuesToTicket();
+        return {ticket: this.ticket};
+    }
+
+    // mapFormValuesToTicket returns a zendesk ticket with AppCall values mapped
+    // 1:1 to a zendesk ticket
+    mapFormValuesToTicket(): void {
+        // app form values that are not to be 1:1 mapped to zendesk fields
+        const omitFields = [AppFieldNames.AdditionalMessage, AppFieldNames.PostMessage];
+
+        // iterate through each field and build the Zendesk ticket
+        Object.keys(this.formValues).forEach((fieldName) => {
+            switch (true) {
+            case omitFields.includes(fieldName):
+                return;
 
-export default getTicketFromForm;
+            // if a form is not defined, selected, or checked in the modal, its
+            // value will be null. continue to next field
+            case !this.formValues[fieldName]:
+                return;
+
+            // field is a custom field
+            case fieldName.startsWith(AppFieldNames.CustomFieldPrefix):
+                this.handleCustomField(fieldName);
```

Comment:
maybe rename to updateTicketCustomField(s) or similar (and similarly for validateField, etc).

Reply:
Please see above referenced changes